src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,965 @@
+     * The era and other fields in {@link ChronoField} are defined by the chronology.
+     *
+     * @return the International fixed chronology, not null
+     */
+    @Override
+    public InternationalFixedChronology getChronology() {
+        return InternationalFixedChronology.INSTANCE;
+    }
+
+    /**
+     * Gets the era applicable at this date.
+     * <p/>
+     * The International fixed calendar system only has one era, 'CE',
+     * defined by {@link InternationalFixedEra}.
+     *
+     * @return the era applicable at this date, not null
+     */
+    @Override
+    public InternationalFixedEra getEra() {
+        return InternationalFixedEra.CE;
+    }
+
+    /**
+     * Returns the length of the month represented by this date.
+     * <p/>
+     * This returns the length of the month in days.
+     * Month lengths do not match those of the ISO calendar system.
+     *
+     * Since Leap Day / Year Day are not part of any month, their 'imaginary' month is of length 1.
+     *
+     * @return the length of the month in days: 28
+     */
+    @Override
+    public int lengthOfMonth() {
+        return month > 0 ? InternationalFixedChronology.DAYS_IN_MONTH : 1;
+    }
+
+    /**
+     * Returns the length of the year represented by this date.
+     * <p/>
+     * This returns the length of the year in days.
+     * Year lengths match those of the ISO calendar system.
+     *
+     * @return the length of the year in days: 365 or 366
+     */
+    @Override
+    public int lengthOfYear() {
+        return InternationalFixedChronology.DAYS_IN_YEAR + (isLeapYear() ? 1 : 0);
+    }
+
+    //-------------------------------------------------------------------------
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate with(final TemporalAdjuster adjuster) {
+        return (InternationalFixedDate) adjuster.adjustInto(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate with(final TemporalField field, final long newValue) {
+        if (field instanceof ChronoField) {
+            ChronoField f = (ChronoField) field;
+            getChronology().range(f).checkValidValue(newValue, f);
+
+            if (f == ChronoField.DAY_OF_MONTH || f == ChronoField.MONTH_OF_YEAR) {
+                if (newValue == 0) {
+                    return createYearDay(getProlepticYear());
+                }
+
+                if (newValue == -1) {
+                    return createLeapDay(getProlepticYear());
+                }
+            }
+
+            int dom = isYearDay() ? 21 : (getCalculatedDayOfMonth() / 7) * 7;
+            int d = day < 1 ? 1 : day % 7;
+            int nval = (int) newValue;
+
+            switch (f) {
+                case DAY_OF_WEEK:
+                case ALIGNED_DAY_OF_WEEK_IN_MONTH:
+                case ALIGNED_DAY_OF_WEEK_IN_YEAR:
+                    if (newValue == 0) {
```

Comment:
There are 2 tests that "move" a date with ALIGNED_DAY_OF_WEEK_IN_YEAR and 0 as argument, lines 616 and 665.

```java
       {2014, 0, 0, ALIGNED_DAY_OF_WEEK_IN_YEAR, 0, 2014, 0, 0}, // line 616
       {2012, -1, -1, ALIGNED_DAY_OF_WEEK_IN_YEAR, 0, 2012, -1, -1}, // line 665
```

I have no problem restricting ALIGNED_DAY_OF_WEEK_IN_YEAR to the range [1, 7], as it is in AbstractDate.

So, the question is, is 0 valid in date.with(ALIGNED_DAY_OF_WEEK_IN_YEAR, 0) ?
Currently it is.

TestDiscordianChronology, line 540 does a very similar positive test
```java
       {2014, 0, 0, ALIGNED_DAY_OF_WEEK_IN_YEAR, 0, 2014, 0, 0},
```

In my book, this spells as: implemented as intended.